platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationFilesPoller.java

```diff
@@ -94,6 +97,7 @@ public void init() throws IOException {
     public void register(@NotNull ChangeListener listener) {
         notNull(listener, "ChangeListener cannot be null");
         changeListener = listener;
+        processExistingConfigurationFiles();
```

Comment:
Yeah, good point. It's unlikely but not impossible. We could always add a wait here (countdown latch or something) to wait until the watch thread has started.